dlt/sources/helpers/requests/api.py

```diff
@@ -0,0 +1,61 @@
+import threading
+from typing import Optional, TYPE_CHECKING
+
+import requests
+
+from dlt.sources.helpers.requests.session import Session
+from dlt.sources.helpers.requests.retry import requests_with_retry
+
+
+_local = threading.local()
+
+
+def _get_session() -> Session:
+    # Manage a thread local session with default settings
+    session: Optional[Session] = getattr(_local, 'session', None)
```

Comment:
ok this is cool. we just need to upgrade this code to use a shared connection pool that is thread safe.